src/lib.rs

```diff
@@ -86,13 +86,46 @@ pub mod store;
 pub mod tests;
 pub mod translate;
 
-pub use data::*;
-pub use executor::*;
-pub use parse_sql::*;
-pub use plan::*;
-pub use result::*;
-pub use store::*;
-pub use translate::*;
+pub mod prelude {
+    #[cfg(feature = "sled-storage")]
+    pub use crate::storages::SledStorage;
 
-pub use glue::Glue;
-pub use storages::*;
+    #[cfg(feature = "memory-storage")]
+    pub use crate::storages::MemoryStorage;
+
+    pub use crate::{
+        data::value::Value,
+        executor::{execute, Payload},
+        glue::Glue,
+        parse_sql::parse,
+        translate::translate,
+    };
+}
+
+pub mod test {
+    #[cfg(feature = "sled-storage")]
+    pub use crate::storages::{sled_storage, SledStorage};
+
+    #[cfg(feature = "memory-storage")]
+    pub use crate::storages::{memory_storage, MemoryStorage};
+
+    #[cfg(feature = "alter-table")]
+    pub use crate::store::AlterTableError;
+
+    #[cfg(feature = "index")]
+    pub use crate::store::IndexError;
+
+    pub use crate::{
+        ast::*,
+        data::{
+            value::{Value::*, ValueError},
+            IntervalError, Literal, LiteralError, RowError,
+        },
+        executor::*,
+        parse_sql::{parse, parse_expr},
+        plan::plan,
+        result::*,
+        store::{GStore, GStoreMut, StoreMut},
+        translate::{translate, translate_expr, TranslateError},
+    };
+}
```

Comment:
Exposing `test` module may sound unclear because we already have `tests` module for providing integration test cases.
What we basically need as an external custom storage maker is the way to use our internal module.
Then, it would be just enough to provide our modules as public.

```rust
pub mod ast;      
pub mod data;     
pub mod executor; 
pub mod parse_sql;
pub mod plan;     
pub mod result;   
pub mod storages; 
pub mod store;    
pub mod tests;    
pub mod translate;
pub mod utils;    
```

Let's provide a single way to access specific module or function.
(Only the exception is `prelude`)

To remove this `test` module, all the test case module imports need to be updated.
However, I think that is necessary task to do.
How do you think?

Reply:
That's reasonable. I also thought the role of my test mod was kinda ambiguous. Thanks for the  suggestion!